Source/SocketIO/Engine/SocketEngine.swift

```diff
@@ -371,6 +371,9 @@ open class SocketEngine : NSObject, URLSessionDelegate, SocketEnginePollable, So
         fastUpgrade = false
         probing = false
         flushProbeWait()
+        if postWait.count != 0 {
```

Comment:
How about a `!.isEmpty`?

Reply:
No problem, I just copied existing code.

Reply:
Or maybe move that check into flush... and unconditionally call it?

Reply:
I'm fine with this way in that case! Would you mind also adding some documentation comments to these methods that explain this failure case? I think those will be helpful when I come back in 6 months and try to refactor and end up breaking it again :).

Reply:
👌 will also change to `!.isEmpty` and test